### PR TITLE
Initial API, fetch data

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 73505: Initial API](https://edu-redmine.sun-asterisk.vn/issues/73505)
## WHAT
- Initial API, fetch data
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/weather/data/repository/source/local/WeatherLocalDataSource.kt [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+class WeatherLocalDataSource : WeatherDataSource.Local {
+    override fun getCityLocal(listener: RequestCompleteListener<MutableList<City>>) {
+        try {
+            val cityList: MutableList<City> = mutableListOf()
+
+            cityList.add(City(HCM, "Hồ Chí Minh", "VN"))
+            cityList.add(City(TDM, "Thủ Dầu Một", "VN"))
+            cityList.add(City(HN, "Hà Nội", "VN"))
+            cityList.add(City(MT, "Manchester", "GB"))
+            cityList.add(City(MC, "Moscow", "RU"))
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/data/repository/source/local/WeatherLocalDataSource.kt]: hard code à em?

Comment (daolq-2712) [reply app/src/main/java/com/example/weather/data/repository/source/local/WeatherLocalDataSource.kt:17]: nên load từ resource ra nha e

**Review by daolq-2712 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/weather/data/repository/source/local/WeatherLocalDataSource.kt (hunk 1 above)
Comment (daolq-2712) [reply app/src/main/java/com/example/weather/data/repository/source/local/WeatherLocalDataSource.kt]: Mình nên load từ resource ra

Comment (ghost) [reply app/src/main/java/com/example/weather/data/repository/source/local/WeatherLocalDataSource.kt]: Dạ chỗ này em tính hard code để test thử thôi á anh, sang phần sau em sẽ load từ resource

Comment (ghost) [reply app/src/main/java/com/example/weather/data/repository/source/local/WeatherLocalDataSource.kt]: Em xóa chỗ hardcode đó rồi anh

**Review by daolq-2712 — APPROVED** [review]
LGTM!

**Pull request merged**